File: problems/3514/solution.cpp
#include <algorithm>
#include <unordered_set>
#include <vector>
using namespace std;

class Solution {
 public:
  int uniqueXorTriplets(vector<int>& nums) {
    int ret = 0;
    vector<bool> level2(2049, false);
    vector<bool> level3(2049, false);
    for (auto& n : nums) {
      for (auto& m : nums) {
        level2[n ^ m] = true;
      }
    }
    for (int i = 0; i < level2.size(); i++) {
      if (!level2[i]) {
        continue;
      }
      for (auto& n : nums) {
        if (!level3[i ^ n]) {
          ret++;
          level3[i ^ n] = true;
        }
      }
    }
    return ret;
  }
};

File: problems/3514/solution.py
from typing import List


class Solution:
    def uniqueXorTriplets(self, nums: List[int]) -> int:
        level2 = [False] * 2049
        level3 = [False] * 2049
        ret = 0

        for n in nums:
            for m in nums:
                level2[n ^ m] = True

        for i in range(len(level2)):
            if not level2[i]:
                continue
            for m in nums:
                if not level3[i ^ m]:
                    level3[i ^ m] = True
                    ret += 1
        return ret
